### PR TITLE
[WIP] Test for all supported HTTP verbs

### PR DESCRIPTION
**Pull request by ewilde**
## TODO
- [x] Wait for release of alpine function with watchdog 0.8.9 =>
- [x] Manual testing
- [x] PR text

## Description
Adds test for invoking functions using "GET", "POST", "PUT", "PATCH", "DELETE"

Part of : https://github.com/openfaas/faas/issues/815

Signed-off-by: Edward Wilde <ewilde@gmail.com>

### CODE REVIEW

File: create-kubernetes-secret.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (ewilde) [inline create-kubernetes-secret.sh]: need to use `bash` `[[` is not part of `sh`

File: Makefile [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 .EXPORT_ALL_VARIABLES:
-secrets:
+secrets-swarm:
 	./create-swarm-secret.sh
+
+secrets-kubernetes:
 	./create-kubernetes-secret.sh
-test: secrets
+
+test-swarm: secrets-swarm clean-swarm
```
Comment (ewilde) [inline Makefile]: separate targets for testing different providers, if you are only running against `swarm` the `create-kubernetes-secret.sh` scripts hangs

File: create-swarm-secret.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (alexellis) [inline create-swarm-secret.sh]: Can we convert the code to run with `sh` instead?

Comment (ewilde) [reply create-swarm-secret.sh]: no idea, why would you want to?

Comment (alexellis) [reply create-swarm-secret.sh]: Not all platforms have bash available by default I.e. Alpine Linux.

**Pull request merged**